homeassistant/components/iammeter/sensor.py

```diff
@@ -77,45 +79,43 @@ async def async_update_data():
     )
     await coordinator.async_refresh()
     entities = []
-    for sensor_name, (row, idx, unit) in api.iammeter.sensor_map().items():
-        serial_number = api.iammeter.serial_number
-        uid = f"{serial_number}-{row}-{idx}"
-        entities.append(IamMeter(coordinator, uid, sensor_name, unit, config_name))
+    for sensor_name, val in api.measurement.items():
+        if sensor_name == "sn":
+            serial_number = val
+            continue
+        if sensor_name in ("Model", "mac"):
+            continue
```

Comment:
It might be better to migrate to using `SensorEntityDescription`s, we should not have to filter out what is received, we should instead only use what we expect.